src/ZoneServer/Commands/ChatCommands.Handlers.cs

```diff
@@ -327,6 +330,118 @@ private CommandResult HandleJump(Character sender, Character target, string mess
 			return CommandResult.Okay;
 		}
 
+		/// <summary>
+		/// Official slash command to hire a pet
+		/// </summary>
+		/// <example>/pethire 3 Pet</example>
+		/// <param name="sender"></param>
+		/// <param name="target"></param>
+		/// <param name="message"></param>
+		/// <param name="command"></param>
+		/// <param name="args"></param>
+		/// <returns></returns>
+		private CommandResult HandlePetHire(Character sender, Character target, string message, string command, Arguments args)
+		{
+			// Since this command is sent via UI interactions, we'll not
+			// use any automated command result messages, but we'll leave
+			// debug messages for now, in case of unexpected values.
+			if (args.Count < 2)
+			{
+				Log.Debug("HandlePetHire: Invalid call by user '{0}': {1}", sender.Username, command);
+				return CommandResult.Okay;
+			}
+
+			if (sender.Companions.HasCompanions)
+				return CommandResult.Okay;
+
+			if (!int.TryParse(args.Get(0), out var petShopId))
+				return CommandResult.InvalidArgument;
+
+			if (!ZoneServer.Instance.Data.CompanionDb.TryFind(petShopId, out var data))
+				return CommandResult.InvalidArgument;
+
+			if (!ZoneServer.Instance.Data.MonsterDb.TryFind(data.ClassName, out var monData))
+				return CommandResult.InvalidArgument;
+
+			var targetNpcName = "Companion Trader";
+			var currentNpcName = sender.Connection.CurrentDialog?.Npc.Name;
+
+			if (string.IsNullOrEmpty(currentNpcName) || !currentNpcName.Contains(targetNpcName))
+				return CommandResult.InvalidArgument;
+
+			//TODO: Decide if we sell pets that don't have a price defined.
+			if (data.Price == 0)
+				return CommandResult.Okay;
+
+			if (sender.Inventory.CountItem(ItemId.Silver) < data.Price)
+			{
+				sender.SystemMessage("OwnerDontHaveSilver");
```

Comment:
Since we decided not to rely on client-side translation, messages should be written out and wrapped in localization calls, instead of using system messages.